src/javaRosa.js

```diff
@@ -1857,8 +1857,8 @@ define([
                                 if (!seen.hasOwnProperty(form)) {
                                     seen[form] = true;
                                     // set default value(s) for this form
-                                    var dkey = name + ":" + dlang + "-" + form;
-                                    item.set(data[dkey] || value || nodeID, form);
+                                    var dval = data[name + ":" + dlang + "-" + form];
+                                    item.set(dval !== undefined ? dval : value, form);
```

Comment:
No, but I didn't like this because it could set the itext value to `null`. Updated.